src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -3670,6 +3670,18 @@ namespace winrt::Microsoft::Terminal::Control::implementation
     void TermControl::PreviewInput(const winrt::hstring& text)
     {
         get_self<ControlCore>(_core)->PreviewInput(text);
+
+        if (!text.empty())
+        {
+            if (auto automationPeer{ FrameworkElementAutomationPeer::FromElement(*this) })
+            {
+                automationPeer.RaiseNotificationEvent(
+                    AutomationNotificationKind::ItemAdded,
+                    AutomationNotificationProcessing::All,
+                    winrt::hstring{ fmt::format(std::wstring_view{ RS_(L"PreviewTextAnnouncement") }, text.c_str()) },

```

Comment:
When you are going from a smart container with knowledge of it length, don't go through `c_str` unnecessarily.

```suggestion
                    winrt::hstring{ fmt::format(std::wstring_view{ RS_(L"PreviewTextAnnouncement") }, std::wstring_view{ text }) },
```

Reply:
you know, you actually don't need to do anything special to pass an hstring as a format arg. It transparently acts like a string view.

```suggestion
                    winrt::hstring{ fmt::format(std::wstring_view{ RS_(L"PreviewTextAnnouncement") }, text) },
```